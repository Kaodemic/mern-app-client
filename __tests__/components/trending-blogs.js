/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import TrendingBlogs from '../../components/trending-blogs';

it('renders ArticleItem', () => {
  const tree = renderer.create(
    <TrendingBlogs
      avatar={'/v1621748084728/nQ7lrJxnS.jpeg'}
      name={'Rahul'}
      userName={'rahul'}
    />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
