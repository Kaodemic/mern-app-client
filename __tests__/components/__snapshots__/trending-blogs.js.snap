// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArticleItem 1`] = `
<div
  className="flex items-start py-3 space-x-2"
>
  <span
    className="text-xl font-semibold opacity-75 text-gray-500"
  >
    01
  </span>
  <a
    className="text-sm font-medium"
    href="/@rahul"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <noscript>
        <img
          alt="Rahul"
          className="rounded-full"
          decoding="async"
          src="/v1621748084728/nQ7lrJxnS.jpeg"
          srcSet="/_next/image?url=%2Fv1621748084728%2FnQ7lrJxnS.jpeg&w=48&q=75 1x, /_next/image?url=%2Fv1621748084728%2FnQ7lrJxnS.jpeg&w=96&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
      <img
        alt="Rahul"
        className="rounded-full"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
    </div>
  </a>
  <div
    className="flex flex-col flex-1 truncate"
  >
    <h4
      className="font-medium leading-snug tracking-tight"
    >
      <a
        href="/@rahul"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Rahul
      </a>
    </h4>
    <a
      className="text-sm block text-gray-500"
      href="/@rahul"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      sayedalesawy.hashnode.dev
    </a>
  </div>
  <button
    className="FollowButton small"
    data-title="Follow blog"
  >
    <svg
      className="w-5 h-5 fill-current"
      viewBox="0 0 384 512"
    >
      <path
        d="M368 224H224V80c0-8.84-7.16-16-16-16h-32c-8.84 0-16 7.16-16 16v144H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h144v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V288h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
      />
    </svg>
  </button>
</div>
`;
